docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1,27 +1,1957 @@
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#bob noComments}
+
+The rest of the test is designed to check that `ALICE` has `BOB`'s coin and vice versa. It starts by calling `next_tx` to make sure the effects of the previous transaction have been committed, before running the necessary checks.
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#finish noComments}
+
+:::tip Additional resources
+
+- [Full source code](https://github.com/MystenLabs/sui/blob/705ee1ed3ce8cfadc4597c6facb6769d7dfb5896/examples/trading/contracts/escrow/sources/shared.move)
+- Guides: [Test Scenario](../../../guides/developer/first-app/build-test#testing-a-package)
+
+:::
+
+### Observability
+
+The `escrow` Move package is now functional: You could publish it on chain and perform trustless swaps by creating transactions. Creating those transactions requires knowing the IDs of `Locked`, `Key`, and `Escrow` objects.
+
+`Locked` and `Key` objects are typically owned by the transaction sender, and so can be queried through the Sui RPC, but `Escrow` objects are shared, and it is useful to be able to query them by their sender and recipient (so that users can see the trades they have offered and received).
+
+Querying `Escrow` objects by their sender or recipient will require custom indexing, and to make it easy for the indexer to spot relevant transactions you will add the following **events** to `escrow.move`:
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#struct=EscrowCreated,EscrowSwapped,EscrowCancelled noComments}
+
+Functions responsible for various aspects of the escrow's lifecycle emit these events. The custom indexer can then subscribe to transactions that emit these events and process only those, rather than the entire chain state:
+
+<details>
+<summary>
+`emit` events included in functions from `shared.move`
+</summary>
+{@inject: examples/trading/contracts/escrow/sources/shared.move#use=sui::event}
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=create,swap,return_to_sender noTitle noComments}
+</details>
+
+:::tip Additional resources
+
+- Concepts: [Events](../../../concepts/events)
+
+:::
+
+
+### Next steps
+
+Well done. You have written the Move package! 🚀
+
+To turn this into a complete dApp, you need to create a frontend. However, for the frontend to be updated, it has to listen to the blockchain as escrows are made and swaps are fulfilled.
+
+To achieve this, in the next step you create an indexing service.
+
+## Backend indexer {#backend}
```

Comment:
(Follow-up, from previous PR):

> Explanation of the schema and indexer could also be broken up into stages -- explain the initial set-up, then how to index one kind of event end-to-end, then add subsequent events (which require less explanation), to avoid having to justify all parts of the schema up-front without it being clear what they would be used for.
>
> We could take this to the next level -- i.e. only build out the locked.move Move module, its indexing, and the UI for locking and unlocking objects before moving on to the Escrow part, but I don't think that's worthwhile -- I think it's easy to justify the interface to each high level part (Move package, Backend, API, Front-end) and then break it down at that level.
>
> This can be a follow-up.